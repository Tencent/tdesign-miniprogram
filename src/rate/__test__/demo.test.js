/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import path from 'path';
import simulate from 'miniprogram-simulate';

const mapper = [
  'action',
  'base',
  'color',
  'count',
  'custom',
  'custom-prefix',
  'show-text',
  'size',
  'special',
  'un-filled',
];

describe('Rate', () => {
  mapper.forEach((demoName) => {
    it(`Rate ${demoName} demo works fine`, () => {
      const id = load(path.resolve(__dirname, `../../rate/_example/${demoName}/index`), demoName);
      const container = simulate.render(id);
      container.attach(document.createElement('parent-wrapper'));
      expect(container.toJSON()).toMatchSnapshot();
    });
  });
});
