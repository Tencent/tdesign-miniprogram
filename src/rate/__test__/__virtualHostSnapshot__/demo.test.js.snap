// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate Rate action demo works fine 1`] = `
<action>
  <wx-view
    class="demo-desc"
  >
    只可选全星时
  </wx-view>
  <wx-view
    class="demo-rate"
  >
    <wx-view
      class="demo-rate__title"
    >
      点击或滑动
    </wx-view>
    <t-rate
      value="{{3.5}}"
      bind:change="onChange"
    />
  </wx-view>
  <wx-view
    class="demo-desc"
  >
    只可选半星时
  </wx-view>
  <wx-view
    class="demo-rate"
  >
    <wx-view
      class="demo-rate__title"
    >
      点击或滑动
    </wx-view>
    <t-rate
      allowHalf="{{true}}"
      defaultValue="{{3}}"
    />
  </wx-view>
</action>
`;

exports[`Rate Rate base demo works fine 1`] = `
<base>
  <wx-view
    class="demo-rate"
  >
    <wx-view
      class="demo-rate__title"
    >
      实心评分
    </wx-view>
    <t-rate
      value="{{3}}"
      bind:change="onChange"
    />
  </wx-view>
</base>
`;

exports[`Rate Rate color demo works fine 1`] = `
<color>
  <wx-view
    class="demo-rate custom-color"
  >
    <wx-view
      class="demo-rate__title"
    >
      填充评分
    </wx-view>
    <t-rate
      allowHalf="{{true}}"
      defaultValue="{{3}}"
    />
  </wx-view>
  <wx-view
    class="demo-rate"
  >
    <wx-view
      class="demo-rate__title"
    >
      线描评分
    </wx-view>
    <t-rate
      color="#00a870"
      defaultValue="{{3}}"
      icon="{{
        Array [
          "star-filled",
          "star",
        ]
      }}"
    />
  </wx-view>
</color>
`;

exports[`Rate Rate count demo works fine 1`] = `
<count>
  <wx-view
    class="demo-rate"
  >
    <wx-view
      class="demo-rate__title"
    >
      自定义评分数量
    </wx-view>
    <t-rate
      count="{{3}}"
      value="{{2}}"
      bind:change="onChange"
    />
  </wx-view>
</count>
`;

exports[`Rate Rate custom demo works fine 1`] = `
<custom>
  <wx-view
    class="demo-rate"
  >
    <wx-view
      class="demo-rate__title"
    >
      自定义评分
    </wx-view>
    <t-rate
      icon="thumb-up"
      value="{{3}}"
      bind:change="onChange"
    />
  </wx-view>
</custom>
`;

exports[`Rate Rate show-text demo works fine 1`] = `
<show-text>
  <wx-view
    class="demo-rate"
  >
    <wx-view
      class="demo-rate__title"
    >
      带描述评分
    </wx-view>
    <t-rate
      data-index="{{0}}"
      showText="{{true}}"
      texts="{{
        Array [
          "1分",
          "2分",
          "3分",
          "4分",
          "5分",
        ]
      }}"
      value="{{3}}"
      bind:change="onChange"
    />
  </wx-view>
  <wx-view
    class="demo-rate"
  >
    <wx-view
      class="demo-rate__title"
    >
      带描述评分
    </wx-view>
    <t-rate
      data-index="{{1}}"
      showText="{{true}}"
      value="{{3}}"
      bind:change="onChange"
    />
  </wx-view>
  <wx-view
    class="demo-rate"
  >
    <wx-view
      class="demo-rate__title"
    >
      带描述评分
    </wx-view>
    <t-rate
      data-index="{{2}}"
      showText="{{true}}"
      value="{{0}}"
      bind:change="onChange"
    />
  </wx-view>
</show-text>
`;

exports[`Rate Rate size demo works fine 1`] = `
<size>
  <wx-view
    class="demo-rate"
  >
    <wx-view
      class="demo-rate__title"
    >
      大尺寸 24
    </wx-view>
    <t-rate
      data-index="{{0}}"
      value="{{3}}"
      bind:change="onChange"
    />
  </wx-view>
  <wx-view
    class="demo-rate"
  >
    <wx-view
      class="demo-rate__title"
    >
      小尺寸 20
    </wx-view>
    <t-rate
      data-index="{{1}}"
      size="{{20}}"
      value="{{3}}"
      bind:change="onChange"
    />
  </wx-view>
</size>
`;

exports[`Rate Rate special demo works fine 1`] = `
<special>
  <wx-view
    class="demo-rate"
  >
    <wx-view
      class="rate-wrapper"
    >
      <t-rate
        size="30"
        texts="{{
          Array [
            "非常糟糕",
            "有些糟糕",
            "可以尝试",
            "可以前往",
            "推荐前往",
          ]
        }}"
        value="{{4}}"
        bind:change="onChange"
      />
    </wx-view>
    <wx-view
      class="desc desc--active"
    >
      可以前往
    </wx-view>
  </wx-view>
</special>
`;

exports[`Rate Rate un-filled demo works fine 1`] = `
<un-filled>
  <wx-view
    class="demo-rate"
  >
    <wx-view
      class="demo-rate__title"
    >
      请点击评分
    </wx-view>
    <t-rate
      value="{{3}}"
      bind:change="onChange"
    />
  </wx-view>
</un-filled>
`;
